File: p1/Deepspace/WeaponType.rb
#encoding:utf-8
module Deepspace
    module WeaponType
        class Type 
            def initialize(_power)
                @power=_power
            end

            def power() 
                @power
            end 
            def to_s()
                case @power
                when 2.0
                    "LASER"
                when 3.0
                    "MISSILE"
                when 4.0
                    "PLASMA"
                end
            end 
        end
        LASER = Type.new(2.0)
        MISSILE = Type.new(3.0)
        PLASMA = Type.new(4.0) 
    end
    #a = WeaponType::LASER
    #b = WeaponType::MISSILE
    #c = WeaponType::PLASMA
    #puts a,b,c 
    #puts a.power,b.power,c.power
end 

File: p2/deepspace/src/deepspace/WeaponType.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Enum.java to edit this template
 */
package deepspace;

/**
 *
 * @author cristobalmer
 */
public enum WeaponType {
    
    LASER(2f), MISSILE(3f), PLASMA(4f); 
    //atributo instancia privado
    private final float power;
    //visibilidad de paquete
    WeaponType(float _power) { 
        power = _power;
    }
     //visibilidad de paquete
    float getPower(){
        return power;
    }
   
}
